### DISCUSSION TITLE
Binary Search using C

### DISCUSSION THREAD

**Discussion by pzth** [community/community#67561] [Programming Help] [answered]
### Body

I am trying to implement binary search in an array using C language but I guess I am bumping into some issues I don't know the issue really because there are no errors in the console.  my code is below.

```c
#include <stdio.h>

int main(){
    int a[50], i, low, high, mid, n, x;

    printf("Enter size of array: ");
    scanf("%d", n);
    printf("\n Enter %d elements in sorted order: \n", n);
    for(i=0; i<n; i++){
        scanf("%d", &a[i]);
    }
    printf("\nEnter value to be searched: ");
    scanf("%d", x);
    low = 0;
    high = n-1;
    while(low <= high){
        mid=(low + high)/2;
        if(x < a[mid]){
            high = mid-1;
        }
        else if(x > a[mid]){
            low = mid+1;
        } else{
        printf("\nSuccessful search\nElement found at index[%d]", mid);
        getch();
        }
    }

    printf("\nElement is not found");
    getch();
}
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by preetsuthar17** [comment, marked as answer]
@pzth I have reviewed your code and I see some mistakes in the code.

The main mistake is when using `scanf()` you're not using & before the variable name, and this is the reason why you're isn't working like it is supposed to.

fixed code will be,

```c
    printf("Enter size of array: ");
    scanf("%d", &n);
```

👆 Here usage of `&` before taking value for the variable from a user.
And the same mistake here,

```c
    printf("\nEnter value to be searched: ");
    scanf("%d", &x);
```

The issue of `&`. Try implementing these changes and you will get the expected output.

Thank you,

Best regards,
Preet

**Reply by pzth** [reply]
Hey Preet, 
Thank you for your time. The issue seems to be resolved :D

Thanks,
pzth
